src/types/mod.rs

```diff
@@ -35,28 +39,51 @@ impl From<KeyId> for String {
 }
 
 impl KeyId {
+    /// Encodes the [`KeyId`] in Base64.
+    ///
+    /// The internal counter is represented in big-endian format and then
+    /// encoded using Base64.
     pub fn to_base64(self) -> String {
         crate::utilities::base64_encode(self.0.to_be_bytes())
     }
 }
 
-/// Error type describing failures that can happen when we try decode or use a
+/// Error type for failures that may occur when decoding or using a
 /// cryptographic key.
 #[derive(Error, Debug)]
 pub enum KeyError {
-    #[error("Failed decoding a public key from base64: {}", .0)]
+    /// Failed to correctly decode a public key that was encoded in Base64.
+    #[error("Failed to decode a public key from Base64: {0}")]
     Base64Error(#[from] base64::DecodeError),
-    #[error("Failed to decode a private key from base64: {}", .0)]
+
+    /// Failed to correctly decode a private key that was encoded in Base64.
+    #[error("Failed to decode a private key from Base64: {0}")]
     Base64PrivateKey(#[from] base64ct::Error),
+
+    /// The Base64 encoded key does not contain the expected number of bytes.
     #[error(
-        "Failed decoding {key_type} key from base64: \
+        "Failed to decode {key_type} key from Base64: \
         Invalid number of bytes for {key_type}, expected {expected_length}, got {length}."
     )]
-    InvalidKeyLength { key_type: &'static str, expected_length: usize, length: usize },
+    InvalidKeyLength {
+        /// The type of key being decoded.
+        key_type: &'static str,
+        /// The expected length of the key.
+        expected_length: usize,
+        /// The actual length of the key.
+        length: usize,
+    },
+    /// Unable to decompress the curve point `r` from an [`Ed25519PublicKey`].
+    ///
+    /// For more details, see [RFC 8032, Section 5.1.3](https://www.rfc-editor.org/rfc/rfc8032.html#section-5.1.3).
     #[error(transparent)]
     Signature(#[from] SignatureError),
-    /// At least one of the keys did not have contributory behaviour and the
-    /// resulting shared secret would have been insecure.
+
+    /// One or more keys lacked contributory behavior in the Diffie-Hellman
+    /// operation, resulting in an insecure shared secret.
+    ///
+    /// For more details on contributory behavior please refer to the
+    /// [`x25519_dalek::SharedSecret::was_contributory()`] method.
     #[error("At least one of the keys did not have contributory behaviour")]
```

Comment:
Should probably change the error to use the same phrasing?

Reply:
d9e9a70c6ddf3e8bd6b2f6021533806e4ce47aa3